core/core.go

```diff
@@ -283,8 +323,37 @@ func NewTextile(ctx context.Context, conf Config) (*Textile, error) {
 	}
 	var opts []grpc.ServerOption
 	if conf.Hub {
+		var powStub *grpcdynamic.Stub
+		var healthServiceDesc *desc.ServiceDescriptor
+		var netServiceDesc *desc.ServiceDescriptor
+		var ffsServiceDesc *desc.ServiceDescriptor
+		var walletServiceDesc *desc.ServiceDescriptor
+		if conf.AddrPowergateAPI != "" {
+			if powStub, err = createPowStub(conf.AddrPowergateAPI); err != nil {
+				return nil, err
+			}
+			if healthServiceDesc, err = createServiceDesciptor("health/rpc/rpc.proto", healthServiceName); err != nil {
+				return nil, err
+			}
+			if netServiceDesc, err = createServiceDesciptor("net/rpc/rpc.proto", netServiceName); err != nil {
+				return nil, err
+			}
+			if ffsServiceDesc, err = createServiceDesciptor("ffs/rpc/rpc.proto", ffsServiceName); err != nil {
+				return nil, err
+			}
+			if walletServiceDesc, err = createServiceDesciptor("wallet/rpc/rpc.proto", walletServiceName); err != nil {
```

Comment:
Each of these service descriptors allows us to look up methods by name to be executed against the stub